### ISSUE TITLE
KernelOp doen't work in scipy with RK45

### ISSUE THREAD

**Issue by erikvansebille** [OceanPARCELS/parcels#88]
While working on Issue #80, I found a strange bug where Kernel additions (`AdvectionRK45 + UpdateP`) don't work in scipy mode when using the AdvectionRK45 kernel.

The AdvectionRK4 and AdvectionEE kernels work, and all works in JIT; out of the 6 possible permutations of modes and advection kernels, it's only RK45/scipy that doesn't work

I've created a branch fix_scipyRK45KernelOps (commit ae88f19) to highlight this error. @mlange05, could you have a look at it when you have time?

**Comment by erikvansebille** [comment]
In fact, looking a little deeper in it, AdvectionRK45 also doesn't seem to work in jit mode. While running with jit doesn't throw an error, the trajectories are actually wrong.

See for example the following terminal output
```
$ python tests/test_peninsula.py jit -m RK4 -v -p 1
Initial particle positions:
P(0.0270, 0.0315)[p=0.00000, p_start=-3.096488]
Peninsula: Advecting 1 particles for 1 day, 0:00:00
Compiled MyParticleAdvectionRK4UpdateP ==> /var/folders/r2/8593q8z93kd7t4j9kbb_f7p00000gn/T/parcels-501/efc436f109ce8e1080a7178258ed2154.so
Final particle positions:
P(0.6971, 0.0416)[p=-3.09723, p_start=-3.096488]

$ python tests/test_peninsula.py jit -m RK45 -v -p 1
Initial particle positions:
P(0.0270, 0.0315)[p=0.00000, p_start=-3.096488]
Peninsula: Advecting 1 particles for 1 day, 0:00:00
Compiled MyParticleAdvectionRK45UpdateP ==> /var/folders/r2/8593q8z93kd7t4j9kbb_f7p00000gn/T/parcels-501/cc976f8cdf8b094181058feb95443e1d.so
Final particle positions:
P(4.7692, 0.3769)[p=0.00000, p_start=-3.096488]
```
The first call (`-m RK4`) yields correct Final particle positions (with longitude around 0.7)
The second call (`-m RK45`), however, leads to a particle actually exiting the domain (with longitude more than 4.7!)
The particle goes _much_ too fast in the RK45 case...

When reducing `interval` to `delta(minutes=5)`, so that it is equal to `dt`, there is no change to the `-m RK4` run, but the longitude in the `-m RK45` run gets even larger, clearly showing that the AdvectionRK45 implementation is broken

Something must have gone wrong when reworking the AdvectionRK45 kernel, because @Abobie worked well with the pre-merged version.

**Comment by erikvansebille** [comment]
RK45 now works in JIT mode in PR #105. However, Scipy doesn't work yet. 

I've now nailed it down to the `return KernelOp.FAILURE` statement on line 76 of `advection.py`. If this statement is changed to `return 0`, then Scipy works, but JIT doesn't. 

So the `KernelOp` is not implemented in Scipy it seems. This is in fact also the error when `python examples/example_stommel.py scipy -m RK45` is called:
```NameError: global name 'KernelOp' is not defined```

**Comment by erikvansebille** [comment]
Fixed in PR #105
